src/providers/google/gdrive-document.ts

```diff
@@ -0,0 +1,207 @@
+        syncPosition.breakId = syncPosition.futureBreakId;
+        syncPosition.futureBreakId = undefined;
+
+        return syncPosition;
+    }
+
+    protected setNextPosition(
+        syncPosition: SyncHandlerPosition,
+        serverResponse: GaxiosResponse<drive_v3.Schema$FileList>
+    ): SyncHandlerPosition {
+        if (!syncPosition.futureBreakId && serverResponse.data.files.length) {
+            syncPosition.futureBreakId = serverResponse.data.files[0].id;
+        }
+
+        if (_.has(serverResponse, "data.nextPageToken")) {
+            // Have more results, so set the next page ready for the next request
+            syncPosition.syncMessage = `Batch complete (${this.config.batchSize}). More results pending.`;
+            syncPosition.thisRef = serverResponse.data.nextPageToken;
+        } else {
+            syncPosition.syncMessage = "Stopping. No more results.";
+            syncPosition = this.stopSync(syncPosition);
+        }
+
+        return syncPosition;
+    }
+
+    protected async buildResults(
+        drive: drive_v3.Drive,
+        auth: OAuth2Client,
+        serverResponse: GaxiosResponse<drive_v3.Schema$FileList>,
+        breakId: string,
+        breakTimestamp?: string
+    ): Promise<SchemaDocument[]> {
+        const results: SchemaDocument[] = [];
+        for (const file of serverResponse.data.files) {
+            const fileId = file.id;
+
+            if (fileId == breakId) {
+                const logEvent: SyncProviderLogEvent = {
+                    level: SyncProviderLogLevel.DEBUG,
+                    message: `Break ID hit (${breakId})`
+                }
+                this.emit('log', logEvent)
+                break;
+            }
+
+            const modifiedTime = file.modifiedTime || "Unkown";
+            
+            if (breakTimestamp && modifiedTime < breakTimestamp) {
+                const logEvent: SyncProviderLogEvent = {
+                    level: SyncProviderLogLevel.DEBUG,
+                    message: `Break timestamp hit (${breakTimestamp})`
+                }
+                this.emit('log', logEvent)
+                break;
+            }
+
+            const title = file.name || "No title";
+            const link = file.webViewLink || "No link";
+            const mimeType = file.mimeType || "Unknown";
+            const type = GoogleDriveHelpers.getDocumentTypeFromMimeType(mimeType);
+            const thumbnail = file.thumbnailLink || "No thumbnail";
+            const size = await GoogleDriveHelpers.getFileSize(drive, file.id)
+            const textContent = await GoogleDriveHelpers.extractIndexableText(drive, file.id, mimeType, auth);
+            
+            results.push({
+                _id: this.buildItemId(fileId),
+                name: title,
+                type: type,
+                size: size,
+                uri: link,
+                icon: thumbnail,
+                contentText: textContent,
+                sourceId: file.id,
```

Comment:
Does google provide `inserted` timestamp that is different from `modified`?

Reply:
> Does google provide `inserted` timestamp that is different from `modified`?

Added `createdTime`